internal/env/parse.go

```diff
@@ -237,3 +237,32 @@ func ParseFloatSlice(val string) ([]float64, error) {
 
 	return floats, nil
 }
+
+func ParseDict(v interface{}) (*Dict, error) {
+	if v == nil {
+		return nil, nil
+	}
+
+	var d = make(Dict)
+
+	switch val := v.(type) {
+	case map[string]interface{}:
+		for k, v := range val {
+			switch v.(type) {
+			case string:
+				s, err := ParseString(v)
+				if err != nil {
+					return nil, err
+				}
+
+				d[k] = *s
```

Comment:
would it make sense to check if the `v` if of type  `map[string]interface{}` and if so, call ParseDict on it as well?

Reply:
i dont think `v` can be of type `map[string]interface{}` or can it?

Reply:
@iwpnd I'm pretty sure it can be. I don't believe TOML has a max limit of nesting. For the tegola config we don't really have much nesting, but it seems to me the ParseDict() function should likely account for all primitives as well as a `map[string]interface{}` (which would just call ParseDict() again)